image-rs/src/snapshots/occlum/unionfs.rs

```diff
@@ -124,18 +143,48 @@ impl Snapshotter for Unionfs {
             .file_name()
             .ok_or(anyhow!("Unknown error: file name parse fail"))?;
 
-        // For mounting trusted UnionFS at runtime of occlum,
-        // you can refer to https://github.com/occlum/occlum/blob/master/docs/runtime_mount.md#1-mount-trusted-unionfs-consisting-of-sefss.
-        // "c7-32-b3-ed-44-df-ec-7b-25-2d-9a-32-38-8d-58-61" is a hardcode key used to encrypt or decrypt the FS currently,
-        // and it will be replaced with dynamic key in the near future.
+        let random_key = generate_random_key();
+        let new_key_path = Path::new("/new_key");
+        fs::create_dir_all(new_key_path)?;
+        std::fs::write(&PathBuf::from(&new_key_path.join("key.txt")), &random_key)?;
```

Comment:
could this be:
```suggestion
        std::fs::write(&PathBuf::from(&keys_mount_path.join("key.txt")), &random_key)?;
```
and moved to the end to replace `fs::copy()`